apollo-router/src/protocols/websocket.rs

```diff
@@ -607,21 +638,35 @@ mod tests {
                 socket.close().await.unwrap();
             });
 
-            Ok(res)
-        }
+            Ok::<_, Infallible>(res)
+        };
 
         let app = Router::new().route("/ws", get(ws_handler));
-        let server = Server::bind(&socket_addr).serve(app.into_make_service());
-        server.await.unwrap();
+        let server = Server::bind(&"127.0.0.1:0".parse().unwrap()).serve(app.into_make_service());
```

Comment:
The test previously allowed the user to specify the address, so it was more flexible. Do we want to remove that?

I don't have strong opinions on this, but wonder if it was because the test could be containerised or because knowing which port you were on allowed you to snoop the tests with wireshark or a proxy or something like that...

Reply:
the test forced a specific port, which was kind of inconvenient, what i do now is return the actual port once it s been bound.
I agree it would be harder to wireshark, happy to put an Option<port> in the function if it helps, wdyt?

Reply:
Options sounds like the perfect solution.

Reply:
fixed in [8917ded](https://github.com/apollographql/router/pull/3562/commits/8917dedd4b1fe63f81f4525bbdabba986e3ad2b0)